client/kafka/kafka.go

```diff
@@ -112,8 +127,13 @@ func (ap *AsyncProducer) ActiveBrokers() []string {
 // Close shuts down the producer and waits for any buffered messages to be
 // flushed. You must call this function before a producer object passes out of
 // scope, as it may otherwise leak memory.
-func (ap *AsyncProducer) Close() error {
-	err := ap.prod.Close()
+func (ap *KafkaProducer) Close() error {
+	var err error
```

Comment:
i am a bit concerned with the if branches... would it make sense to go a bit more on the functional approach, and create a close func in the KafkaProducer struct , that would be generated in the Create ?
`close func() error`

Reply:
This issue will be addressed by returning a separate object with its own diversified `Close()`, see other conversation thread.